### PR TITLE
Fix sky_shell test flakiness caused by CPU throttling

### PR DESCRIPTION
**Pull request by tvolkert**
When a machine is heavily loaded, it can severely delay the time it
takes for the OS to start a process once it's asked to do so. Our
5 second timeout that we were giving the test process to connect
to the test harness seemed like plenty of time, were it not for the
fact that the test process itself was not being started in time when
CPU throttling was in effect.

This change updates the test timeout to begin counting only once the
test process has been started. We keep the original timeout in play
in the event that the test process *never* starts up for some reason,
but we up that timeout value to 5 minutes.

### CODE REVIEW

**Comment by tvolkert** [comment]
@Hixie

File: packages/flutter_tools/lib/src/test/flutter_platform.dart [hunk 1]
```diff
@@ -20,7 +20,19 @@ import '../dart/package_map.dart';
...
+/// once the process has been started.
 const Duration _kTestStartupTimeout = const Duration(seconds: 5);
+
+/// The timeout we give the operating system to start the test process
+/// once it's been instructed to do so. When the OS is under severe load,
+/// it throttles processes but will start them eventually.
+const Duration _kTestProcessTimeout = const Duration(minutes: 5);
+
+/// Message logged by the test process to signal that it has started.
+const String _kStartTimeoutTimerMessage = 'msg:startTime';
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/test/flutter_platform.dart]: I would just make this a human-readable string.

File: packages/flutter_tools/lib/src/test/flutter_platform.dart [hunk 2]
```diff
@@ -115,18 +127,32 @@ class FlutterPlatform extends PlatformPlugin {
         processToKill: process, // This kills the subprocess whether coverage is enabled or not.
       );
 
+      Completer<Null> timeout = new Completer<Null>();
+
       // Pipe stdout and stderr from the subprocess to our printStatus console.
-      _pipeStandardStreamsToConsole(process);
+      _pipeStandardStreamsToConsole(process, handleLine: (String line) {
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/test/flutter_platform.dart]: put the string comparison logic in _pipeStandardStreamsToConsole alongside the "error: Unable to read Dart source" code. Just pass a callback for "main started" that does the future stuff.

File: packages/flutter_tools/lib/src/test/flutter_platform.dart (hunk 1 above)
Comment (tvolkert) [inline packages/flutter_tools/lib/src/test/flutter_platform.dart]: Done

File: packages/flutter_tools/lib/src/test/flutter_platform.dart (hunk 2 above)
Comment (tvolkert) [inline packages/flutter_tools/lib/src/test/flutter_platform.dart]: Done

**Comment by Hixie** [comment]
LGTM

**Pull request merged**